File: js/examples/basic_audit.js
/**
 * Базовый пример: аудит одного сайта.
 * node examples/basic_audit.js https://example.com
 */
import { SitePravo } from '../sitepravo.js';

const url = process.argv[2] || 'https://example.com';

console.log(`Запуск аудита: ${url}\n`);

const client = new SitePravo();
const report = await client.audit(url, {
  onProgress: (status, elapsed) => {
    process.stdout.write(`\r  ⏳ ${status} (${Math.round(elapsed / 1000)}с)  `);
  },
});

console.log(`\n\n${'='.repeat(60)}`);
console.log(`URL:   ${report.url}`);
console.log(`Grade: ${report.grade}   Score: ${report.score}/100`);
console.log(`Всего: ${report.totalFindings} находок`);
console.log(`  critical: ${report.summary.critical ?? 0}`);
console.log(`  high:     ${report.summary.high ?? 0}`);
console.log(`  medium:   ${report.summary.medium ?? 0}`);
console.log(`  low:      ${report.summary.low ?? 0}`);
console.log(`${'='.repeat(60)}\n`);

if (report.criticalFindings.length > 0) {
  console.log('🚨 КРИТИЧЕСКИЕ НАРУШЕНИЯ:');
  for (const f of report.criticalFindings) {
    console.log(`\n  [${f.id}] ${f.title}`);
    console.log(`  Где: ${f.where}`);
    console.log(`  Риск: ${f.whyRisk.slice(0, 200)}`);
    console.log(`  Исправить: ${f.fix.slice(0, 200)}`);
  }
} else {
  console.log('✅ Критических нарушений не найдено');
}

if (report.highFindings.length > 0) {
  console.log(`\n⚠️  ВЫСОКИЙ ПРИОРИТЕТ (${report.highFindings.length} шт.):`);
  for (const f of report.highFindings.slice(0, 5)) {
    console.log(`  - [${f.category}] ${f.title}`);
  }
}

File: python/examples/basic_audit.py
#!/usr/bin/env python3
"""
Базовый пример: запуск аудита и вывод критических находок.
"""
import sys
sys.path.insert(0, '..')

from sitepravo import SitePravo

def main():
    url = sys.argv[1] if len(sys.argv) > 1 else "https://example.com"

    print(f"Запуск аудита: {url}")
    client = SitePravo()
    report = client.audit(url)

    print(f"\n{'='*60}")
    print(f"URL:   {report.url}")
    print(f"Grade: {report.grade}  Score: {report.score}/100")
    print(f"Всего находок: {report.total_findings}")
    print(f"  critical: {report.summary.get('critical', 0)}")
    print(f"  high:     {report.summary.get('high', 0)}")
    print(f"  medium:   {report.summary.get('medium', 0)}")
    print(f"  low:      {report.summary.get('low', 0)}")
    print(f"Проверено: {report.checked_at}")
    print(f"{'='*60}\n")

    if report.critical_findings:
        print("🚨 КРИТИЧЕСКИЕ НАРУШЕНИЯ:")
        for f in report.critical_findings:
            print(f"\n  [{f.id}] {f.title}")
            print(f"  Где: {f.where}")
            print(f"  Риск: {f.why_risk[:200]}")
            print(f"  Исправить: {f.fix[:200]}")
    else:
        print("✅ Критических нарушений не найдено")

    if report.high_findings:
        print(f"\n⚠️  ВЫСОКИЙ ПРИОРИТЕТ ({len(report.high_findings)} шт.):")
        for f in report.high_findings[:5]:
            print(f"  - [{f.category}] {f.title}")


if __name__ == "__main__":
    main()
